pkg/network/usm/ebpf_gotls.go

```
@@ -124,58 +125,82 @@ type GoTLSProgram struct {
 	registry *utils.FileRegistry
 }
 
-// Static evaluation to make sure we are not breaking the interface.
-var _ subprogram = &GoTLSProgram{}
+var goTLSSpec = &protocols.ProtocolSpec{
+	Maps: []*manager.Map{
+		{Name: offsetsDataMap},
+		{Name: goTLSReadArgsMap},
+		{Name: goTLSWriteArgsMap},
+		{Name: connectionTupleByGoTLSMap},
+	},
+	Probes: []*manager.Probe{
+		{
+			ProbeIdentificationPair: manager.ProbeIdentificationPair{
+				EBPFFuncName: connReadProbe,
+			},
+		},
+		{
+			ProbeIdentificationPair: manager.ProbeIdentificationPair{
+				EBPFFuncName: connReadRetProbe,
+			},
+		},
+		{
+			ProbeIdentificationPair: manager.ProbeIdentificationPair{
+				EBPFFuncName: connWriteProbe,
+			},
+		},
+		{
+			ProbeIdentificationPair: manager.ProbeIdentificationPair{
+				EBPFFuncName: connWriteRetProbe,
+			},
+		},
+		{
+			ProbeIdentificationPair: manager.ProbeIdentificationPair{
+				EBPFFuncName: connCloseProbe,
+			},
+		},
+	},
+}
 
-func newGoTLSProgram(c *config.Config, sockFD *ebpf.Map) *GoTLSProgram {
-	if !c.EnableGoTLSSupport {
-		return nil
-	}
+func newGoTLSProgramProtocolFactory(m *manager.Manager, sockFDMap *ebpf.Map) protocols.ProtocolFactory {
+	return func(c *config.Config) (protocols.Protocol, error) {
+		if !c.EnableGoTLSSupport {
+			return nil, nil
+		}
 
-	if !http.HTTPSSupported(c) {
-		log.Errorf("goTLS not supported by this platform")
-		return nil
-	}
+		if !http.HTTPSSupported(c) {
+			log.Errorf("goTLS not supported by this platform")
+			return nil, nil
+		}
 
-	if !c.EnableRuntimeCompiler && !c.EnableCORE {
-		log.Errorf("goTLS support requires runtime-compilation or CO-RE to be enabled")
-		return nil
-	}
+		if !c.EnableRuntimeCompiler && !c.EnableCORE {
+			log.Errorf("goTLS support requires runtime-compilation or CO-RE to be enabled")
+			return nil, nil
```

return the error instead of logging it